widget/entry_validation.go

```diff
@@ -29,10 +29,15 @@ func (e *Entry) SetOnValidationChanged(callback func(error)) {
 // SetValidationError manually updates the validation status until the next input change
 func (e *Entry) SetValidationError(err error) {
 	e.validationError = err
+	if e.Validator == nil {
+		return
+	}
 
-	if e.Validator != nil {
-		e.validationStatus.Refresh()
+	if err != e.validationError && e.onValidationChanged != nil {
```

Comment:
`err != e.validationError` will never be true because of line 31

Reply:
Good point. I don't know why that slipped in.